lib/foreman/renderer.rb

```diff
@@ -93,16 +93,21 @@ def indent(count)
       prefix + text.gsub(/\n/, "\n#{prefix}")
     end
 
+    # accepts either template object or plain string
     def unattended_render(template, template_name = nil)
-      content = template.respond_to?(:template) ? template.template : template
       template_name ||= template.respond_to?(:name) ? template.name : 'Unnamed'
+      raise ::Foreman::Exception.new(N_("Template '%s' is either missing or has invalid organization or location"), template_name) if template.nil?
+      content = template.respond_to?(:template) ? template.template : template
       allowed_variables = ALLOWED_VARIABLES.reduce({}) do |mapping, var|
         mapping.update(var => instance_variable_get("@#{var}"))
       end
       allowed_variables[:template_name] = template_name
       render_safe content, ALLOWED_HELPERS, allowed_variables
     end
-    alias_method :pxe_render, :unattended_render
+
+    def pxe_render(template, template_name = nil)
+      ::Foreman::Deprecation.deprecation_warning("1.11", "method pxe_render will be removed, use unattended_render instead")
```

Comment:
1.12, according to http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches#Deprecation-warnings